renderdoc/core/bit_flag_iterator.h

```diff
@@ -0,0 +1,43 @@
+/******************************************************************************
+* The MIT License (MIT)
+*
+* Copyright (c) 2018-2019 Baldur Karlsson
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+******************************************************************************/
+#pragma once
+
+template <typename B, typename F, typename S>
+struct BitFlagIterator
+{
+  F mask;
```

Comment:
I don't feel very strongly about this, but my initial reaction before reading the code was that `mask` was actually a mask applied to the bits being iterated, so you could choose to e.g. only iterate flags in the upper bits or something. Perhaps it could be renamed to 'flags' instead since this is named a flag iterator?

In either event I do think this should be a private member, otherwise there's a temptation to think it still stores the original set of flags being iterated, when it's continually masking out the lowest bit on increment.